KeystreamGenerator.rb

```diff
@@ -0,0 +1,138 @@
+class KeystreamGenerator
+
+  attr_reader :keystream_length, :deck
```

Comment:
Instead of making `deck` a reader, I'd use a memoized method, like:

```ruby
def deck
  @deck ||= build_deck
end
```

This way, the first time `deck` is called, it will then call `build_deck`, store the results in `@deck`, then return contents. Any subsequent calls will just return the value of `@deck`, so it's sort of "cached" after the first method call.